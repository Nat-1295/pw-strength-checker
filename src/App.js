import React, { useEffect, useState } from 'react';
import { useDebounce } from 'use-debounce';
import './styles/main.css';
import { Layout, Row, Col } from 'antd';

import { getPwStrength } from './api/PWStrengthAPI';
import TextField from "./components/Field/TextField";
import ScoreBar from './components/Scorebar/ScoreBar';
import Message from "./components/Message/Message";
import { scoreBarData } from "./constant/constant";


const App = () => {
  const [pwData, setPwData] = useState(null);
  const [password, setPassword] = useState('');
  const debouncedPassword = useDebounce(password, 200);

  useEffect(async () => {
    if (!debouncedPassword[0]) {
      setPwData(null);
      return;
    }
    await getPwStrength(debouncedPassword.shift()).then(data => {
      setPwData(data);
    });
  }, [debouncedPassword[0]]);


  return (
    <Layout>
      <Row align="middle">
        <Col className="container" xs={12} align="middle">
          <Col xs={9} style={{ fontSize: '20px', marginBottom: '8%' }}><b>Is your password strong enough?</b></Col>
          <Col xs={20}><TextField setPassword={setPassword} /></Col>
          <Row xs={24} justify="center">
            {scoreBarData.map((item) => <Col xs={4} key={item.id}><ScoreBar {...{ ...pwData, ...item, password }} /></Col>)}
          </Row>
          <Col xs={24}>
            <Message {...pwData} />
          </Col>
        </Col>
      </Row>
    </Layout>
  );
};

export default App;
