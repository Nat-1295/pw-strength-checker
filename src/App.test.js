/* eslint-disable no-console */
import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';


beforeAll(() => {
  Object.defineProperty(window, "matchMedia", {
    writable: true,
    value: jest.fn().mockImplementation(query => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
      fetch: jest.fn(),
    })),
  });
});

const renderApp = () => render(<App />);

describe('generate a snapshot', () => {
  it("should match the snapshot", () => {
    const {
      container: { firstChild },
    } = renderApp();
    expect(firstChild).toMatchSnapshot();
  });

  it('title text should be visible', () => {
    renderApp();
    expect(screen.getByText("Is your password strong enough?")).toBeTruthy();
  });
});
