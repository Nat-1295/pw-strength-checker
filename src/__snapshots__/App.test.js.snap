// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate a snapshot should match the snapshot 1`] = `
<section
  class="ant-layout"
>
  <div
    class="ant-row ant-row-middle"
  >
    <div
      align="middle"
      class="ant-col container ant-col-xs-12"
    >
      <div
        class="ant-col ant-col-xs-9"
        style="font-size: 20px; margin-bottom: 8%;"
      >
        <b>
          Is your password strong enough?
        </b>
      </div>
      <div
        class="ant-col ant-col-xs-20"
      >
        <span
          class="ant-input-affix-wrapper ant-input-password"
          style="border: 1px solid;"
        >
          <input
            action="click"
            class="ant-input"
            placeholder="Type a password"
            type="password"
            value=""
          />
          <span
            class="ant-input-suffix"
          >
            <span
              class="ant-input-password-icon"
            >
              Show
            </span>
          </span>
        </span>
      </div>
      <div
        class="ant-row ant-row-center"
        xs="24"
      >
        <div
          class="ant-col ant-col-xs-4"
        >
          <div
            class="scorebar"
            style="background: gray;"
          />
        </div>
        <div
          class="ant-col ant-col-xs-4"
        >
          <div
            class="scorebar"
            style="background: gray;"
          />
        </div>
        <div
          class="ant-col ant-col-xs-4"
        >
          <div
            class="scorebar"
            style="background: gray;"
          />
        </div>
        <div
          class="ant-col ant-col-xs-4"
        >
          <div
            class="scorebar"
            style="background: gray;"
          />
        </div>
        <div
          class="ant-col ant-col-xs-4"
        >
          <div
            class="scorebar"
            style="background: gray;"
          />
        </div>
      </div>
      <div
        class="ant-col ant-col-xs-24"
      >
        <div
          class="ant-row ant-row-center"
        >
          <div
            class="ant-space ant-space-vertical"
          >
            <div
              class="ant-space-item"
              style="margin-bottom: 8px;"
            >
              <div
                class="ant-col ant-col-xs-24"
                style="font-size: 13px;"
              >
                 
              </div>
            </div>
            <div
              class="ant-space-item"
            >
              <div
                class="ant-col ant-col-xs-24"
              >
                <b />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`it should load the password strength checker should match the snapshot 1`] = `
<section
  class="ant-layout"
>
  <div
    class="ant-row ant-row-middle"
  >
    <div
      align="middle"
      class="ant-col container ant-col-xs-12"
    >
      <div
        class="ant-col ant-col-xs-9"
        style="font-size: 20px; margin-bottom: 8%;"
      >
        <b>
          Is your password strong enough?
        </b>
      </div>
      <div
        class="ant-col ant-col-xs-20"
      >
        <span
          class="ant-input-affix-wrapper ant-input-password"
          style="border: 1px solid;"
        >
          <input
            action="click"
            class="ant-input"
            placeholder="Type a password"
            type="password"
            value=""
          />
          <span
            class="ant-input-suffix"
          >
            <span
              class="ant-input-password-icon"
            >
              Show
            </span>
          </span>
        </span>
      </div>
      <div
        class="ant-row ant-row-center"
        xs="24"
      >
        <div
          class="ant-col ant-col-xs-4"
        >
          <div
            class="scorebar"
            style="background: gray;"
          />
        </div>
        <div
          class="ant-col ant-col-xs-4"
        >
          <div
            class="scorebar"
            style="background: gray;"
          />
        </div>
        <div
          class="ant-col ant-col-xs-4"
        >
          <div
            class="scorebar"
            style="background: gray;"
          />
        </div>
        <div
          class="ant-col ant-col-xs-4"
        >
          <div
            class="scorebar"
            style="background: gray;"
          />
        </div>
        <div
          class="ant-col ant-col-xs-4"
        >
          <div
            class="scorebar"
            style="background: gray;"
          />
        </div>
      </div>
      <div
        class="ant-col ant-col-xs-24"
      >
        <div
          class="ant-row ant-row-center"
        >
          <div
            class="ant-space ant-space-vertical"
          >
            <div
              class="ant-space-item"
              style="margin-bottom: 8px;"
            >
              <div
                class="ant-col ant-col-xs-24"
                style="font-size: 13px;"
              >
                 
              </div>
            </div>
            <div
              class="ant-space-item"
            >
              <div
                class="ant-col ant-col-xs-24"
              >
                <b />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
